### PR TITLE
[BUG] Fix the bug of query in expr 

### LINKED ISSUE #6767: [Bug] Query bug of in expr
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

table1: 
```
+---------+--------+------+-------+---------+-------+
| Field   | Type   | Null | Key   | Default | Extra |
+---------+--------+------+-------+---------+-------+
| city_id | BIGINT | Yes  | true  | NULL    |       |
| tag_id  | BIGINT | Yes  | true  | NULL    |       |
| dt      | BIGINT | Yes  | true  | NULL    |       |
| num     | BIGINT | Yes  | false | NULL    | NONE  |
+---------+--------+------+-------+---------+-------+

select * from table1 where city_id in ("1100000000000000005");

+---------------------+--------+----------+------+
| city_id             | tag_id | dt       | num  |
+---------------------+--------+----------+------+
| 1100000000000000005 |      1 | 20200102 |    1 |
| 1100000000000000007 |      1 | 20200102 |    1 |
| 1100000000000000006 |      1 | 20200102 |    1 |
+---------------------+--------+----------+------+
```

### What You Expected?

```
select * from table1 where city_id in ("1100000000000000005");
+---------------------+--------+----------+------+
| city_id             | tag_id | dt       | num  |
+---------------------+--------+----------+------+
| 1100000000000000005 |      1 | 20200102 |    1 |
+---------------------+--------+----------+------+
```

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Fix the bug of query in expr 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6767) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yangzhg — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Type.java [hunk 1]
```diff
@@ -1018,6 +1018,13 @@ public static Type getCmpType(Type t1, Type t2) {
                 || (t1ResultType == PrimitiveType.VARCHAR && t2ResultType == PrimitiveType.STRING)) {
             return Type.STRING;
         }
+
+        // int family type and char family type should cast to char family type
+        if ((t1ResultType.isFixedPointType() && t2ResultType.isCharFamily()) ||
+                (t2ResultType.isFixedPointType() && t1ResultType.isCharFamily())) {
+            return t1.isStringType() ?  t1 : t2;
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Type.java:1025]: 数字类型按照字符来排序存在错误的可能。比如1235与12345，按数字递增是1235,12345，按字符串递增就是12345,1235

Comment (HappenLee) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Type.java:1025]: 这个逻辑只会用在in和between in，我测试一下

Comment (HappenLee) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Type.java:1025]: 我重新梳理了这部分的代码，在in和between 里应该是没有问题的

**Review by morningman — COMMENTED** [review]
Is there any unit test that can cover this situation?

**Pull request merged**